specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2020-06-01/imds.json

```diff
@@ -0,0 +1,884 @@
+          "type": "string",
+          "description": "This is the name of the VM."
+        },
+        "offer": {
+          "type": "string",
+          "description": "This is the offer information for the VM image. This value is only present for images deployed from the Azure Image Gallery."
+        },
+        "osType": {
+          "type": "string",
+          "description": "This value indicates the type of OS the VM is running, either Linux or Windows."
+        },
+        "placementGroupId": {
+          "type": "string",
+          "description": "This is the placement group of your Virtual Machine Scale Set."
+        },
+        "plan": {
+          "type": "object",
+          "description": "This contains the data about the plan.",
+          "$ref": "#/definitions/PlanProperties"
+        },
+        "publicKeys": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PublicKeysProperties"
+          },
+          "description": "This is information about the SSH certificate"
+        },
+        "platformFaultDomain": {
+          "type": "string",
+          "description": "This is the fault domain in which the VM."
+        },
+        "platformUpdateDomain": {
+          "type": "string",
+          "description": "This is the update domain in which the VM."
+        },
+        "provider": {
+          "type": "string",
+          "description": "This is the provider of the VM."
+        },
+        "publisher": {
+          "type": "string",
+          "description": "This is the publisher of the VM image."
+        },
+        "resourceGroupName": {
+          "type": "string",
+          "description": "This is the resource group for the VM."
+        },
+        "resourceId": {
+          "type": "string",
+          "description": "This is the fully qualified ID for the VM."
+        },
+        "securityProfile": {
+          "type": "object",
+          "description": "This contains the data about the security profile associated with the VM.",
+          "$ref": "#/definitions/SecurityProfile"
+        },
+        "sku": {
+          "type": "string",
+          "description": "This is the specific SKU for the VM image."
+        },
+        "storageProfile": {
+          "type": "object",
+          "description": "This contains the data about the storage disks associated with the VM.",
+          "$ref": "#/definitions/StorageProfile"
+        },
+        "subscriptionId": {
+          "type": "string",
+          "description": "This is the Azure subscription for the VM."
+        },
+        "tags": {
+          "type": "string",
+          "description": "This is the list of tags for your VM."
+        },
+        "tagsList": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TagsProperties"
+          },
+          "description": "This is the list of tags for your VM formatted as a JSON array for easier programmatic parsing."
```

Comment:
This comment makes me curious - why is it easier to programmatically parse a list (array) of key-value pairs than an listing keys and their corresponding value in an object/map? This generally goes against our API recommendations when you have a unique value constraint (e.g. the `name` property in the key/value pair in this list).

Reply:
The tags string takes all the key-value pairs and does a simple string concatenation with colons and semicolons without escaping anything.

If you have colons and/or semicolons in your tag name or the tag value then you can no longer reliably figure out the end of one tag and the start of the next. `tags` remains for backwards compatibility but is broken.

Reply:
Concerning choice of array vs object map... that may have been an oversight as well but the person who designed that is no longer here.